src/Hl7.Fhir.Specification/Specification/Snapshot/ElementDefnMerger.cs

```diff
@@ -340,6 +340,27 @@ private FhirString deepCopyAndRaiseOnConstraint(FhirString elt)
                     return snap;
             }
 
+            private ElementDefinition.ElementDefinitionBindingComponent mergeBinding(ElementDefinition.ElementDefinitionBindingComponent snap, ElementDefinition.ElementDefinitionBindingComponent diff)
+            {
+                var result = snap;
+                if (!diff.IsNullOrEmpty())
+                {
+                    if (snap.IsNullOrEmpty())
+                    {
+                        result = (ElementDefinition.ElementDefinitionBindingComponent)diff.DeepCopy();
+                        onConstraint(result);
+                    }
+                    else if (!diff.IsExactly(snap))
+                    {
+                        snap.StrengthElement = mergePrimitiveAttribute(snap.StrengthElement, diff.StrengthElement);
+                        snap.DescriptionElement = mergePrimitiveAttribute(snap.DescriptionElement, diff.DescriptionElement);
+                        snap.ValueSet = mergeComplexAttribute(snap.ValueSet, diff.ValueSet);
+                        onConstraint(result);
```

Comment:
Check whether this onCronstraint is needed, because it is also called in the functions  `mergePrimitiveAttribute` and `mergeComplexAttribute`.

Reply:
OnConstraint() is indeed needed to mark changed on the BackboneElement.